src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -3691,57 +3711,58 @@ export class StudyViewPageStore
+                );
+            });
+
+            if (!_.isEmpty(sampleIdentifiersFilterSets)) {
+                filters.sampleIdentifiers = _sampleIdentifiers;
             } else {
-                filters.sampleIdentifiers = this.queriedSampleIdentifiers.result;
+                if (_.isEmpty(this.queriedSampleIdentifiers.result)) {
+                    filters.studyIds = this.queriedPhysicalStudyIds.result;
+                } else {
+                    filters.sampleIdentifiers = this.queriedSampleIdentifiers.result;
+                }
             }
-        }
 
-        filters.alterationFilter = ({
-            // select all CNA types
-            copyNumberAlterationEventTypes: {
-                [CopyNumberEnrichmentEventType.AMP]: true,
-                [CopyNumberEnrichmentEventType.HOMDEL]: true,
-            },
-            // select all mutation types
-            mutationEventTypes: {
-                [MutationEnrichmentEventType.any]: true,
-            },
-            structuralVariants: null,
-            includeDriver: this.driverAnnotationSettings.includeDriver,
-            includeVUS: this.driverAnnotationSettings.includeVUS,
-            includeUnknownOncogenicity: this.driverAnnotationSettings
-                .includeUnknownOncogenicity,
-            includeUnknownTier: this.driverAnnotationSettings
-                .includeUnknownTier,
-            includeGermline: this.includeGermlineMutations,
-            includeSomatic: this.includeSomaticMutations,
-            includeUnknownStatus: this.includeUnknownStatusMutations,
-            tiersBooleanMap: this.selectedDriverTiersMap.isComplete
-                ? this.selectedDriverTiersMap.result!
-                : {},
-        } as unknown) as AlterationFilter;
-
-        return filters as StudyViewFilter;
-    }
+            filters.alterationFilter = ({
+                // select all CNA types
+                copyNumberAlterationEventTypes: {
+                    [CopyNumberEnrichmentEventType.AMP]: true,
+                    [CopyNumberEnrichmentEventType.HOMDEL]: true,
+                },
+                // select all mutation types
+                mutationEventTypes: {
+                    [MutationEnrichmentEventType.any]: true,
+                },
+                structuralVariants: null,
+                includeDriver: this.driverAnnotationSettings.includeDriver,
+                includeVUS: this.driverAnnotationSettings.includeVUS,
+                includeUnknownOncogenicity: this.driverAnnotationSettings
+                    .includeUnknownOncogenicity,
+                includeUnknownTier: this.driverAnnotationSettings
+                    .includeUnknownTier,
+                includeGermline: this.includeGermlineMutations,
+                includeSomatic: this.includeSomaticMutations,
+                includeUnknownStatus: this.includeUnknownStatusMutations,
+                tiersBooleanMap: this.selectedDriverTiersMap.isComplete
```

Comment:
we should avoid worry about isComplete.   if it's possible that isComplete is false here then we have a bigger problem, and that's the problem we should solve.

Reply:
this logic is already present in original code, we should create an issue to address this, I guess?